src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -320,6 +320,19 @@ public static bool IsOnDeserializationImplementation([NotNullWhen(returnValue: t
                 method.IsImplementationOfInterfaceMethod(null, iDeserializationCallback, "OnDeserialization");
         }
 
+        public static bool IsSerializationConstructor([NotNullWhen(returnValue: true)] this IMethodSymbol? method, INamedTypeSymbol? serializationInfoType, INamedTypeSymbol? streamingContextType)
+            => method.IsConstructor() &&
+                method.Parameters.Length == 2 &&
+                method.Parameters[0].Type.Equals(serializationInfoType) &&
+                method.Parameters[1].Type.Equals(streamingContextType);
+
+        public static bool IsGetObjectData([NotNullWhen(returnValue: true)] this IMethodSymbol? method, INamedTypeSymbol? serializationInfoType, INamedTypeSymbol? streamingContextType)
+            => method?.Name == "GetObjectData" &&
+                method.ReturnsVoid &&
+                method.Parameters.Length == 2 &&
+                method.Parameters[0].Type.Equals(serializationInfoType) &&
+                method.Parameters[1].Type.Equals(streamingContextType);
```

Comment:
This isn't taking into account accessibility nor parameter names. Do we want to be strict?

Also, is it worth creating a `WellKnownMethods` static class and move all the specific construct there?

Reply:
Can this extension method also be used here: https://github.com/dotnet/roslyn-analyzers/blob/5be2c0cf63a36b43a2736587c02c37a83dfa41e1/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/ParameterValidationAnalysis/ParameterValidationAnalysis.ParameterValidationDataFlowOperationVisitor.cs#L287-L306

Reply:
And also here: https://github.com/dotnet/roslyn-analyzers/blob/5be2c0cf63a36b43a2736587c02c37a83dfa41e1/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SerializationRulesDiagnosticAnalyzer.cs#L200-L206